src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -593,16 +597,23 @@ internal void ThrowIfNotOpenable(bool needToUncompress, bool needToLoadIntoMemor
         private CheckSumAndSizeWriteStream GetDataCompressor(Stream backingStream, bool leaveBackingStreamOpen, EventHandler onClose)
         {
             // stream stack: backingStream -> DeflateStream -> CheckSumWriteStream
-
-            // we should always be compressing with deflate. Stored is used for empty files, but we don't actually
-            // call through this function for that - we just write the stored value in the header
-            Debug.Assert(CompressionMethod == CompressionMethodValues.Deflate);
-
-            Stream compressorStream = _compressionLevel.HasValue ?
-                new DeflateStream(backingStream, _compressionLevel.Value, leaveBackingStreamOpen) :
-                new DeflateStream(backingStream, CompressionMode.Compress, leaveBackingStreamOpen);
+            Debug.Assert(CompressionMethod == CompressionMethodValues.Deflate || CompressionMethod == CompressionMethodValues.Stored);
 
             bool isIntermediateStream = true;
+            Stream compressorStream;
+            if (!_compressionLevel.HasValue)
+            {
+                _compressionLevel = CompressionMode.Compress;
```

Comment:
`CompressionLeve`l != `CompressionMode`

Can't assign one to another. I am assuming this should be `CompressionLevel.NoCompression`?

Reply:
Oh I'm sorry it auto updated the pull request. I'm currently adding a switch statement there

Reply:
OK, I think with the 25e3f1f commit I should have addressed most of your feedback. The test does pass locally but the CI test is probably still failing. I run `..\..\build .` from inside the `\src\System.IO.Compression` directory.